### PR TITLE
CID-2029: Add webhook and token validation

### PR DESCRIPTION
**Pull request by henriqamaral**

### CODE REVIEW

File: src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationService.kt [hunk 1]
```diff
@@ -6,17 +6,21 @@ import net.leanix.vsm.gitlab.broker.connector.domain.GitLabAssignment
...
 import net.leanix.vsm.gitlab.broker.shared.exception.InvalidToken
 import net.leanix.vsm.gitlab.broker.shared.exception.OrgNameValidationFailed
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Component
 
 @Component
 class ValidationService(
-    private val gitlabClientProvider: GitlabClientProvider
+    private val gitlabClientProvider: GitlabClientProvider,
+    @Value("\${leanix.gitlab.webhook-url}") private val gitlabWebhookUrl: String,
```
Comment (ahmed-ali-55) [inline src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationService.kt:15]: Should we make `gitlabWebhookUrl: String?`

File: src/test/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationServiceTest.kt [hunk 2]
```diff
@@ -75,14 +75,34 @@ class ValidationServiceTest {
...
+
+        verify(exactly = 0) { gitlabClient.getCurrentUser() }
+        verify(exactly = 1) { gitlabClient.getAllGroups() }
+    }
+
+    private fun setupService(webhookUrl: String = "webhook-url") {
+        validationService = ValidationService(gitlabFeignClientProvider, webhookUrl)
+        validationService!!.messageSource = messageSource
+        validationService!!.loggingService = loggingService
+    }
```
Comment (ahmed-ali-55) [inline src/test/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationServiceTest.kt:107]: can we move this to `@BeforeEach`??

**Review by ahmed-ali-55 — APPROVED** [review]
Couple of comments. Rest looks good!

File: src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationService.kt (hunk 1 above)
Comment (henriqamaral) [reply src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationService.kt:15]: We don't need it, if you check here if the user doesn't provide the url, we set it as blank:
https://github.com/leanix/vsm-gitlab-broker/blob/d25c925a0b55ce556039c671b83683d44c6a7824/src/main/resources/application.yaml#L18

Comment (ahmed-ali-55) [reply src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationService.kt:15]: Ah, okay!

File: src/test/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationServiceTest.kt (hunk 2 above)
Comment (henriqamaral) [reply src/test/kotlin/net/leanix/vsm/gitlab/broker/connector/application/ValidationServiceTest.kt]: I had moved to two nested classes, one to test with webhook url and another to test without it.

**Pull request merged**